model-server/src/main/kotlin/org/modelix/model/server/handlers/ModelReplicationServer.kt

```diff
@@ -85,7 +85,7 @@ class ModelReplicationServer(val repositoriesManager: RepositoriesManager) {
             call.respondText(storeClient.generateId("clientId").toString())
         }
         get("server-id") {
-            call.respondText(repositoriesManager.getServerId())
+            call.respondText(repositoriesManager.maybeInitAndGetSeverId())
```

Comment:
Why is this a lazy operation on first request and not something done on application start?

Reply:
I guess, that there is no specific reason for this.
The requirement seams just to be that the same `server-id` is returned by the server.
This can be done by initializing it eagerly or lazy.

With my change in `KeyValueLikeModelServer::init` I do initialize it eagerly, because I deemed the code more robust.
If I want to do it lazy in `KeyValueLikeModelServer`, I need to create separate endpoints or conditions for the `SERVER_ID_KEY`, `LEGACY_SERVER_ID_KEY`, `LEGACY_SERVER_ID_KEY2`.
And everybody changing `maybeInitAndGetSeverId` would need to update those special cases.

Reply:
Maybe it would be good to put these explanations into comments or the commit message so the next one touching this piece of code knows why this way was chosen and not another one.

Reply:
Done

Reply:
That's a lot better to understand later on this way!